src/analysis/retail/priest/shadow/modules/talents/Archon/EnergyCompression.tsx

```diff
@@ -0,0 +1,68 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { Options } from 'parser/core/Module';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import { calculateEffectiveDamage } from 'parser/core/EventCalculateLib';
+import Enemies from 'parser/shared/modules/Enemies';
+
+import TALENTS from 'common/TALENTS/priest';
+import SPELLS from 'common/SPELLS';
+
+import Events, { DamageEvent } from 'parser/core/Events';
+
+import { ARCHON_ENERGY_COMPRESSION_MULTIPLIER } from '../../../constants';
+
+class EnergyCompression extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  protected enemies!: Enemies;
+
+  damageHalo = 0;
+  insanityHalo = 0;
+
+  damageEnergyCompression = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.ENERGY_COMPRESSION_TALENT);
+
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.SHADOW_HALO_DAMAGE),
+      this.onHalo,
+    );
+  }
+
+  onHalo(event: DamageEvent) {
+    this.damageHalo += event.amount;
+
+    //Damage from Energy Compression
+    if (this.selectedCombatant.hasTalent(TALENTS.ENERGY_COMPRESSION_TALENT)) {
+      this.damageEnergyCompression += calculateEffectiveDamage(
+        event,
+        ARCHON_ENERGY_COMPRESSION_MULTIPLIER,
+      );
+    }
```

Comment:
this event handler only runs if `this.active` is true, so you don't need to check `hasTalent` again here.
```suggestion
    //Damage from Energy Compression
    this.damageEnergyCompression += calculateEffectiveDamage(
      event,
      ARCHON_ENERGY_COMPRESSION_MULTIPLIER,
    );
```

Reply:
Good catch, I've changed it to so that the event handler always runs, because halo is always talented even with the other hero choice node.